### PR TITLE
docs: Describe custom pipeline deployment #197

### PR DESCRIPTION
**Pull request by Oleksandr123234**
# Describe custom pipeline deployment

## Description
This page provides comprehensive guidelines on how to adjust the deployment logic to cater users' needs.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [*] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

## Checklist:
- [*] I have performed a self-review of my own code
- [*] I have commented my code, particularly in hard-to-understand areas
- [*] I have made corresponding changes to the documentation
- [*] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [*] New and existing unit tests pass locally with my changes
- [*] Pull Request contain one commit. I squash my commits

### CODE REVIEW

File: docs/operator-guide/customize_deployment.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Customize Deployment
+
+Sometimes additional steps are required during application deployment. With the customization of the KubeRocketCI deployment, it's possible to override the deployment logic, which can be useful, for example, if you need to spin up and configure a database before deploying the main application, remove old information from the environment, redeploy the application, and initiate post-deployment configuration, tests, and so on.
```
Comment (NikolayMarusenko) [inline docs/operator-guide/customize_deployment.md:3]: rephrase

File: docs/operator-guide/customize_deployment.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+This is where custom deployment comes into play. This page provides comprehensive guidelines on how to adjust the deployment logic to cater your needs.
+
+## Deploy Custom Pipeline
+
+To customize your deployment pipeline, follow the steps below:
+
```
Comment (NikolayMarusenko) [inline docs/operator-guide/customize_deployment.md:9]: maybe we can provide scheme how we use chain tt, pipeline, task?

File: docs/operator-guide/customize_deployment.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+2. Create the custom pipeline with your custom logic. Refer to the custom pipeline [example](https://github.com/epam/edp-tekton/blob/master/charts/pipelines-library/templates/pipelines/cd/deploy.yaml) for more details.
+
+3. Apply the created manifest files in the `edp` namespace.
+
+4. In the `Create stage` window of the KubeRocketCI portal, select the added trigger template in the corresponding window:
+
+  !![Select trigger template](../assets/operator-guide/select_trigger_template.png "Select trigger template")
+
+## Deploy Custom Pipeline to Remote Cluster (Optional)
```
Comment (NikolayMarusenko) [inline docs/operator-guide/customize_deployment.md:27]: Why do we have this separate section? I see that here you just need to remember that if you remote the cluster, then don't forget the secret and nothing more

File: docs/operator-guide/customize_deployment.md (hunk 1 above)
Comment (Oleksandr123234) [reply docs/operator-guide/customize_deployment.md]: Done

File: docs/operator-guide/customize_deployment.md (hunk 2 above)
Comment (Oleksandr123234) [reply docs/operator-guide/customize_deployment.md:21]: Done

File: docs/operator-guide/customize_deployment.md (hunk 3 above)
Comment (Oleksandr123234) [reply docs/operator-guide/customize_deployment.md]: Done

File: docs/operator-guide/customize_deployment.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# Customize Deployment
+
+When deploying applications into envirnments, it might be necessary to automate some post-deployment procedures, such as testing, configuring, removing old information from the environment, etc. For these purposes, the custom deployment feature was implemented.
```
Comment (SergK) [inline docs/operator-guide/customize_deployment.md:3]: we also support pre-deployment customization, let's just say about customization of deployment in general

File: docs/operator-guide/customize_deployment.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Overall, the custom pipeline creation involves the following steps:
+
+```mermaid
+graph LR;
+    A(Create TriggerTemplate resource) --> B(Create Pipeline resource) --> C(Deploy custom environment)
+```
+
+1. **Create TriggerTemplate resource** -  On this step, we create the `TriggerTemplate` custom resource that will appear as an option in the in the environment stage creation menu.
+2. **Create Pipeline** - On this step, we create another important custom resource called `Pipeline` that complements the trigger template. This resource contains all the tasks to perform within the custom pipeline.
```
Comment (SergK) [inline docs/operator-guide/customize_deployment.md:17]: 'another important' please remove

Comment (Oleksandr123234) [reply docs/operator-guide/customize_deployment.md]: Done

File: docs/operator-guide/customize_deployment.md (hunk 4 above)
Comment (Oleksandr123234) [reply docs/operator-guide/customize_deployment.md]: Done

**Review by SergK — APPROVED** [review]

**Pull request merged**